cmd/empire/factories.go

```diff
@@ -227,3 +229,21 @@ func newHBReporter(key, env string) (reporter.Reporter, error) {
 	// since it thinks MultiReporter is a struct literal.
 	return append(reporter.MultiReporter{}, empire.DefaultReporter, r), nil
 }
+
+// Auth provider =======================
+
+func newAuthProvider(c *cli.Context) (dockerauth.AuthProvider, error) {
+	provider := dockerauth.NewMultiAuthProvider()
+	provider.AddProvider(dockerauth.NewEcrAuthProvider(ecr.New(newConfigProvider(c))))
+
+	dockerConfigFile, err := os.Open(c.String(FlagDockerAuth))
+	if err == nil {
+		defer dockerConfigFile.Close()
+		dockerConfigProvider, err := dockerauth.NewDockerConfigAuthProvider(dockerConfigFile)
+		if err == nil {
```

Comment:
Absolutely. I just think that should be handled with the flag. If the `--docker.auth` flag is an empty string, then we shouldn't add a `dockerConfigAuthProvider` to the chain, instead of ignoring the error. Mostly to prevent misconfiguration (e.g. I really want to use my .dockercfg but I put it in the wrong place).

Basically, I think we should have this:

```go
if fname := c.String(FlagDockerAuth); fname != "" {
        f, err := os.Open(fname)
        if err != nil {
                return nil, err
        }

        p, err := dockerauth.NewDockerConfigAuthProvider(f)
        if err != nil {
                return nil, err
        }

        provider.AddProvider(p)
}
```